native/components/message-markdown.react.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import * as SimpleMarkdown from 'simple-markdown';
+
+import { getMarkdownStyles } from '../themes/markdown-styles';
+import rules from './rules.react';
+
+type Props = {|
+  darkStyle: boolean,
+  children: string,
+|};
+function MessageMarkdown(props: Props) {
+  const style = React.useMemo(() => {
+    return getMarkdownStyles(props.darkStyle ? 'dark' : 'light');
+  }, [props.darkStyle]);
+
+  const customRules = React.useMemo(() => rules(style), [style]);
+
+  const parser = React.useMemo(() => SimpleMarkdown.parserFor(customRules), [
+    customRules,
+  ]);
+
+  const parse = React.useMemo(() => {
+    // we want to parse childen as one block, instead of inline
+    // to be interpreted as a one block, simple-markdown rules require \n\n at the end
+    // we're adding this manually, and specyfying inline as false, so block-parsing is forced
```

Comment:
Thanks for adding all these comments! Two style nits that I'd like you to apply to all of the comments:

1. Let's try to keep line length to a maximum of 80 characters. Prettier doesn't handle enforcing this for code comments, so we have to do it manually
2. This one matters a lot less, but while we're at it let's capitalize the first letter in all of the sentences

Before I merge this PR I'll probably go through all of the comments and reword them a little bit to be clearer.